src/helpers/files.js

```diff
@@ -11,12 +13,41 @@ const TEST_ROUTE = /(|\/)\[[^/]+?](\/|\.html|$)/
 
 const isDynamicRoute = (route) => TEST_ROUTE.test(route)
 
-exports.moveStaticPages = async ({ netlifyConfig, target, i18n, failBuild }) => {
+const stripLocale = (rawPath, locales = []) => {
+  const [locale, ...segments] = rawPath.split('/')
+  if (locales.includes(locale)) {
+    return segments.join('/')
+  }
+  return rawPath
+}
+
+const matchMiddleware = (middleware, filePath) =>
+  middleware.includes('') ||
```

Comment:
I don't have enough context to understand this check. Can you share an example of the shape `middleware` can take?

Reply:
Ah, never mind. I think I figured it out by looking at the tests.

Reply:
It's an array of paths, relative to the site root, that indicate directories which have middleware. Next.js applies middleware to any path below that. The leading slash has been stripped. There are some examples in the unit tests. An empty string indicates root-level middleware thata applies to everything